python from PyQt5.QtWidgets import QDialog, QMessageBox from PyQt5 import uic import os import stomp
class SitrepSendForm(QDialog): def init(self, parent=None): super().init(parent) uic.loadUi(os.path.join(os.path.dirname(file), '../forms/SitrepSendForm.ui'), self) self.buttonSend.clicked.connect(self.send_message) self.buttonCancel.clicked.connect(self.reject)
def send_message(self): from_addr = self.editFrom.text() to_addr = self.editTo.text() subject = self.editSubject.text() message = self.textMessage.toPlainText() full_message = f"От: {from_addr}\nКому: {to_addr}\nТема: {subject}\n\n{message}" try: conn = stomp.Connection([('localhost', 61613)]) conn.connect(wait=True) conn.send(destination="/queue/sitrep", body=full_message) conn.disconnect() QMessageBox.information(self, "Успех", "SITREP отправлен через ActiveMQ.") self.accept() except Exception as e: QMessageBox.critical(self, "Ошибка", f"Ошибка при отправке сообщения:\n{e}")